Author: Matt McClinch <mattmcclinch@gmail.com>
Description: fix #288383: Crash when inserting notes in voice 2 when first rest is deleted
Origin: upstream

--- a/libmscore/input.cpp
+++ b/libmscore/input.cpp
@@ -154,13 +154,14 @@ Segment* InputState::nextInputPos() cons
       Measure* m = _segment->measure();
       Segment* s = _segment->next1(SegmentType::ChordRest);
       for (; s; s = s->next1(SegmentType::ChordRest)) {
-            if (s->element(_track) || s->measure() != m) {
-                  if (s->element(_track)) {
-                        if (s->element(_track)->isRest() && toRest(s->element(_track))->isGap())
-                              continue;
-                        }
-                  return s;
+            if (s->element(_track)) {
+                  if (s->element(_track)->isRest() && toRest(s->element(_track))->isGap())
+                        m = s->measure();
+                  else
+                        return s;
                   }
+            else if (s->measure() != m)
+                  return s;
             }
       return 0;
       }
